src/Digitick/Foundation/Collection/AbstractObjectSet.php

```diff
@@ -124,100 +116,96 @@ public function containsAll(InterfaceCollection $elementCollection)
     }
 
     /**
-     * Check if an element exists in storage
-     *
-     * @param $element
-     * @return mixed
+     * @inheritdoc
      */
     public function contains($element)
     {
         return $this->storageArray->contains($element);
     }
 
     /**
-     * Remove all elements in storage from given list
-     *
-     * @param InterfaceCollection $elementCollection
+     * @inheritdoc
+     * TODO : Probleme SplObjectStorage::removeAll demande un SplObjectStorage mais on y passe un InterfaceCollection ce qui n'est rien pour lui.
      */
     public function removeAll(InterfaceCollection $elementCollection)
```

Comment:
I think we can implement the method. It's different from clear(); removeAll(InterfaceCollection $elementCollection) only remove objects from the given input collection.

The **SplObjectStorage** has the following method :
`public void detach ( object $object )`
We should loop through the input collection and delete each object present in the storageArray property.